lib/features/popup-menu/ReplaceMenuProvider.js

```diff
@@ -461,18 +464,24 @@ ReplaceMenuProvider.prototype._getLoopCharacteristicsHeaderEntries = function(ta
   var translate = this._translate;
 
   function toggleLoopEntry(event, entry) {
-    var newLoopCharacteristics = getBusinessObject(target).loopCharacteristics;
 
+    // remove
     if (entry.active) {
-      newLoopCharacteristics = undefined;
-    } else {
-      if (isUndefined(entry.options.isSequential) || !newLoopCharacteristics
-      || !is(newLoopCharacteristics, entry.options.loopCharacteristics)) {
-        newLoopCharacteristics = self._moddle.create(entry.options.loopCharacteristics);
-      }
+      self._modeling.updateProperties(target, { loopCharacteristics: undefined });
+      return;
+    }
+
+    const currentLoopCharacteristics = target.businessObject.get('loopCharacteristics'),
+          newLoopCharacteristics = self._moddle.create(entry.options.loopCharacteristics);
 
-      newLoopCharacteristics.isSequential = entry.options.isSequential;
+    // copy old properties
+    if (currentLoopCharacteristics) {
+      self._moddleCopy.copyElement(currentLoopCharacteristics, newLoopCharacteristics);
     }
+
+    // update `isSequential` property
+    newLoopCharacteristics.set('isSequential', entry.options.isSequential);
```

Comment:
We could if there was only a single object type assigned to `loopCharacteristics`. However, this can be one of two:
* `bpmn:MultiInstanceLoopCharacteristics` with property `isSequential`, or
* `bpmn:StandardLoopCharacteristics` without properties.

Both of them can have Camunda properties which need to be copied if we change the type of the object.

Reply:
Related to https://github.com/bpmn-io/bpmn-js/pull/1679

Reply:
Or rather https://github.com/bpmn-io/bpmn-js/issues/1581